semanticdb-gradle-plugin/src/main/scala/SemanticdbGradlePlugin.scala

```diff
@@ -0,0 +1,402 @@
+package com.sourcegraph.gradle.semanticdb
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.{util => ju}
+
+import scala.jdk.CollectionConverters._
+
+import com.sourcegraph.scip_java.BuildInfo
+import org.gradle.api.DefaultTask
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.api.tasks.scala.ScalaCompile
+
+class SemanticdbGradlePlugin extends Plugin[Project] {
+  override def apply(project: Project) = {
+    val gradle = new GradleVersion(project.getGradle().getGradleVersion())
+    project.afterEvaluate { project =>
+      project.getRepositories().add(project.getRepositories().mavenCentral())
+      project.getRepositories().add(project.getRepositories().mavenLocal())
+
+      val extra = project.getExtensions().getExtraProperties()
+      val extraProperties = extra.getProperties().asScala
+
+      val targetRoot = extra
+        .getProperties()
+        .asScala
+        .getOrElse("semanticdbTarget", project.getBuildDir())
+
+      val javacPluginVersion = BuildInfo.version
+
+      val javacPluginJar = extraProperties
+        .get("javacPluginJar")
+        .map(_.asInstanceOf[String])
+
+      val javacDep = javacPluginJar
+        .map[Object](jar => project.files(jar))
+        .getOrElse(s"com.sourcegraph:semanticdb-javac:${javacPluginVersion}")
```

Comment:
We fallback to javac plugin published to maven if there is no jar specified

Reply:
Could have been a decent inline comment instead of PR comment